agent/rpc/peering/stream_test.go

```diff
@@ -63,10 +63,25 @@ func TestStreamResources_Server_Follower(t *testing.T) {
 		}
 	}()
 
+	expectedMsg := &pbpeering.ReplicationMessage{
+		Payload: &pbpeering.ReplicationMessage_LeaderAddress_{
+			LeaderAddress: &pbpeering.ReplicationMessage_LeaderAddress{
+				Address: "",
+			},
+		},
+	}
+
+	// expect ReplicationMessage_LeaderAddress
 	msg, err := client.Recv()
-	require.Nil(t, msg)
-	require.Error(t, err)
-	require.EqualError(t, err, "rpc error: code = FailedPrecondition desc = cannot establish a peering stream on a follower node")
+	require.NoError(t, err)
+	require.NotEmpty(t, msg)
+	prototest.AssertDeepEqual(t, expectedMsg, msg)
+
+	// expect error
+	msg2, err2 := client.Recv()
+	require.Nil(t, msg2)
+	require.Error(t, err2)
+	require.Contains(t, err2.Error(), "rpc error: code = FailedPrecondition desc = cannot establish a peering stream on a follower node")
```

Comment:
This was a little un-intuitive to me.

As a dialer, we receive the `LeaderAddress` message then the error as a subsequent "message". The more I started at the code (the mock `Recv()` code that is) the more I convinced myself that this is emulating real world scenario and is in fact what we want but please keep me honest if I got it wrong here.

Reply:
changed approach, irrelevant now